clients/privacy-center/common/geolocation.ts

```diff
@@ -26,7 +32,8 @@ export const LOCATION_HEADERS = [
  *
  */
 export const lookupGeolocation = async (
-  req: NextApiRequest
+  req: NextApiRequest,
+  settings: PrivacyCenterClientSettings
```

Comment:
nit: for code cleanliness, I think it's better to have the caller not pass in a settings hash like this... it feels like the business logic is spidering out a lot and not very maintainable here and you'll get some subtle-but-painful logic errors.

Reply:
The alternative is to hoist the `if (settings.***) ...` type calls out of this function, and require the caller be specific and provide the params like `geolocationApiUrl?: string`

Not sure if this makes sense, can explain live!

Reply:
yeah this makes total sense 💯